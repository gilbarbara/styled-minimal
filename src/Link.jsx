import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { getColor, getDimmerColor } from './utils/helpers';
import { variantPropTypes } from './utils/system';

import Box from './Box';

const Link = styled(Box)(props => {
  const { color } = props;
  const currentColor = color || getColor(props);

  return css`
    color: ${currentColor};

    &:visited {
      color: ${getDimmerColor(currentColor)};
    }
  `;
});

Link.displayName = 'Link';

Link.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  href: PropTypes.string.isRequired,
  variant: variantPropTypes,
  ...Box.propTypes,
};

Link.defaultProps = {
  as: 'a',
};

export default Link;
