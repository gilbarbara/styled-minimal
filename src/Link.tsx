import * as React from 'react';
import styled, { css } from 'styled-components';

import { getColor, getDimmerColor } from './utils/helpers';
import { baseStyles, sx, system } from './utils/system';

import { LinkProps } from './types';

export const StyledLink = styled('a')<LinkProps>(
  props => {
    const color = getColor(props);

    return css`
      ${baseStyles};
      align-items: center;
      color: ${color};
      display: inline-flex;

      &:visited {
        color: ${getDimmerColor(color)};
      }
    `;
  },
  sx,
  system,
);

export const Link = React.forwardRef<HTMLAnchorElement, LinkProps>((props, ref) => (
  <StyledLink ref={ref} {...props} />
));

Link.defaultProps = {
  variant: 'primary',
};
