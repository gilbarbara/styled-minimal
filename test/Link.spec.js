import React from 'react';
import { mount } from 'enzyme';

import Link from '../src/Link';

describe('Link', () => {
  const wrapper = mount(<Link href="https://github.com">GitHub</Link>);

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
