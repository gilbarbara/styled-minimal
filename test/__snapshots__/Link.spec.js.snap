// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  color: #2E2E2E;
}

.c0:visited {
  color: #949494;
}

<Link
  as="a"
  href="https://github.com"
>
  <a
    className="c0"
    href="https://github.com"
  >
    GitHub
  </a>
</Link>
`;
